### ISSUE TITLE
`parsevalsum` and `parsevalsum2` probably trigger scalar operations on GPU

### ISSUE THREAD

**Issue by glwagner** [FourierFlows/FourierFlows.jl#128]
The utility functions `parsevalsum` and `parsevalsum2` likely trigger scalar operations. These functions are defined here:

https://github.com/FourierFlows/FourierFlows.jl/blob/3549d0320fa98df9843fdb5dfe08abd450074a4e/src/utils.jl#L147

Some diagnostics in `GeophysicalFlows.TwoDTurb` depend on `parsevalsum`. I think this probably slows them down.

It should be possible to define a slightly different algorithm for computing the sum. The main issue is that `sum` cannot be called on "non-contiguous" data (in memory), which I think `uh[2:end, :]` is, indeed (the first index is the fast index, so `[2:end, :]` requires "skipping" the first element of each column).

Thus we should rewrite these functions to compute `sum` on contiguous data only. I think this is possible by rewriting `parsevalsum`, for example, to:

```julia
"""
    parsevalsum(uh, g)
Returns `real(Σ uh)` on the grid `g`.
"""
function parsevalsum(uh, g::TwoDGrid)
  if size(uh, 1) == g.nkr                # uh is conjugate symmetric
    U = 2 * sum(uh)                      # sum all modes twice
    U -= @views sum(uh[1, :])            # subtract redundant k=1 mode
  else # count every mode once
    U = sum(uh)
  end

  norm = g.Lx*g.Ly/(g.nx^2*g.ny^2) # weird normalization for dft
  return norm*real(U)
end
```

As an aside, we should adopt the practice of always writing explicit `return` statements within functions. It makes them easier to interpret.

**Comment by glwagner** [comment]
We found that the issue is specifically with `@views` usage in parseval sum.

Here's some code which ran on my machine "sverdrup":

```julia
using CuArrays

a = CuArray(rand(128, 128))

@time CuArrays.@sync sum(abs2, a[2:end, :]) # 0.001192 seconds (448 allocations: 16.422 KiB)
@views @time CuArrays.@sync sum(abs2, a[2:end, :]) # 0.188469 seconds (48.79 k allocations: 1.985 MiB)
```

Since we don't need `@views`, this problem is easily solved by not using `@views`.

**Comment by glwagner** [comment]
Closed by #165 .
